syntax = "proto3";

import "google/protobuf/empty.proto";

package orderqueue;

message Order {
  string orderId = 1;
  repeated OrderItem items = 2;
  int32 orderQuantity = 3;
}

message OrderItem {
    string id = 1;
    int32 quantity = 2;
}

message Confirmation {
  bool isSuccess = 1;
  string message = 2;
}

service OrderQueueService {
  rpc Enqueue(Order) returns (Confirmation) {}
  rpc Dequeue(google.protobuf.Empty) returns (Order) {}
}